static/src/javascripts/projects/common/modules/ui/subscription-banner.js

```diff
@@ -0,0 +1,167 @@
+
+const fiveOrMorePageViews = (currentPageViews: number) => currentPageViews >= 5;
+
+const isAustralianEdition = (currentEdition: string) => currentEdition === 'AU';
+
+const closedAt = (lastClosedAtKey: string) =>
+    userPrefs.set(lastClosedAtKey, new Date().toISOString());
+
+const bannerHasBeenAcknowledged = () => {
+    const messageStates = userPrefs.get('messages') || [];
+    messageStates.push(messageCode);
+    userPrefs.set('messages', uniq(messageStates));
+};
+
+const subcriptionBannerCloseActions = () => {
+    closedAt(subsciptionBannerClosedKey);
+    bannerHasBeenAcknowledged();
+};
+
+const onAgree = (): void => {
+    allAdConsents.forEach(_ => {
+        setAdConsentState(_, true);
+    });
+};
+
+const bindCloseHandler = (button, banner, callback) => {
+    if (button) {
+        button.addEventListener('click', () => {
+            callback();
+            banner.remove();
+        });
+    }
+};
+
+const bindClickHandler = (button, callback) => {
+    if (button) {
+        button.addEventListener('click', () => {
+            callback();
+        });
+    }
+};
+
+const trackSubscriptionBannerView = () => {
+    submitViewEvent({
+        component: {
+            componentType: 'ACQUISITIONS_OTHER',
+            id: 'acquisitions-subscription-banner',
+        },
+    });
+};
+
+const trackSubscriptionBannerCtaClick = () => {
+    submitClickEvent({
+        component: {
+            componentType: 'ACQUISITIONS_OTHER',
+            id: 'acquisitions-subscription-banner',
+        },
+    });
+};
+
+const bindSubscriptionClickHandlers = () => {
+    const subscriptionBannercloseButton = document.querySelector(
+        '#js-site-message--subscription-banner__cta-dismiss'
+    );
+    const subscriptionBannerHtml = document.querySelector(
+        '#js-subscription-banner-site-message'
+    );
+
+    const subscriptionBannerCta = document.querySelector(
+        '#js-site-message--subscription-banner__cta'
+    );
+
+    if (subscriptionBannerHtml) {
+        bindCloseHandler(
+            subscriptionBannercloseButton,
+            subscriptionBannerHtml,
+            subcriptionBannerCloseActions
+        );
+        bindClickHandler(
+            subscriptionBannerCta,
+            trackSubscriptionBannerCtaClick
+        );
+    }
+};
+
+const bindConsentClickHandlers = () => {
+    const consentBannerCloseButton = document.querySelector(
+        '.site-message--first-pv-consent__button'
+    );
+    const consentBannerHtml = document.querySelector(
+        '#js-first-pv-consent-site-message'
+    );
+
+    if (consentBannerHtml) {
+        bindCloseHandler(consentBannerCloseButton, consentBannerHtml, onAgree);
+    }
+};
+
+const show: () => Promise<boolean> = async () => {
+    trackFirstPvConsent();
+    trackSubscriptionBannerView();
+    const showConsent = await canShowFirstPvConsent();
```

Comment:
This feels like it should be inside the `canShow()` function.

Reply:
@damien-shaw-guardian you might want to confirm with @guardian/dotcom-platform  but I don't believe we support `async`/`await`, as described here https://github.com/guardian/frontend/pull/16864#issuecomment-303457881

> From what I understand, in order to transpile async functions, we would need to use Babel's transform-async-to-generator plugin. In turn, this requires us to serve a generator runtime (or Babel's polyfill library), which adds a lot of weight to the application.